include/cudaq/Support/TargetConfig.h

```diff
@@ -0,0 +1,190 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include "llvm/Support/YAMLTraits.h"
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace cudaq {
+namespace config {
+/// Flag to enable feature(s) of the unified NVIDIA target.
+// Use bitset so that we can combine different options, e.g., multi-gpu with
+// fp32/64.
+enum TargetFeatureFlag : unsigned {
+  flagsFP32 = 0x0001,
+  flagsFP64 = 0x0002,
+  flagsMgpu = 0x0004,
+  flagsMqpu = 0x0008,
+};
+
+/// @brief Configuration argument type annotation
+// e.g., to support type validation.
+enum ArgumentType { String, Int, UUID, FeatureFlag };
```

Comment:
Recommend using scoped enums here ("enum class") if possible (to avoid enum names leaking into the parent namespace).